activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -107,9 +111,12 @@ def equality_node?(node)
         end
 
         def non_conflicting_binds(other)
-          conflicts = referenced_columns & other.referenced_columns
-          conflicts.map! { |node| node.name.to_s }
-          binds.reject { |attr| conflicts.include?(attr.name) }
+          # Choose all predicates that had bound params to work with them based on index
+          predicates_with_binds = predicates.select { |predicate| equality_node?(predicate) && predicate.right.is_a?(Arel::Nodes::BindParam) }
```

Comment:
Ok, I see now what you meant. If I have something like this:
`Comment.where(body: 'First comment').merge(Comment.where(developer_id: (1..5)))`
My RHS will actually be `#<Arel::Nodes::And:0x007fc335adedc8 @children=[#<Arel::Nodes::BindParam:0x007fc335ae72c0>, #<Arel::Nodes::BindParam:0x007fc335ae7040>]>`

So we basically have a tree here